### ISSUE TITLE
Unclear about this error for resolver errors

### ISSUE THREAD

**Issue by ndarilek** [absinthe-graphql/absinthe#246]
So I'm still trying to get back more than just an error string from resolvers. Currently I have this simple field:

```
    field :get_open_order_count, type: :integer do
      resolve fn(args, %{context: context}) ->
        {:error, %{foo: "bar"}}
      end
    end
```

I get this exception:

```
** (Absinthe.ExecutionError) Resolution function did not return `{:ok, term}`, `{:error, binary}` or `{:error, map | Keyword.t}`
Resolving field: getOpenOrderCount
Resolving on: %{}
Got: {:error, %{foo: "bar"}}
```

Is `%{foo: "bar"}` not a map? `is_map` says it is. I'm a bit confused about the nature of that error message. Is it asking for a map *or* something else, a map concatenated with something else, or...? I'm fairly new to Elixir, this corner of the API isn't documented anywhere I've seen, and I can't believe there is no way to return an error message with a map-like value. :)

Thanks.

**Comment by bruce** [comment]
I'm sorry, the spec got out of sync with recent changes.

Per the GraphQL specification, an error message is required for a field error. For that reason, the shape you want here is one of:

- `{:error, String.t}`, where the string is the message
- `{:error, Keyword.t | map}`, where a value for `:message` must be provided. Any additional values are fine, just should be JSON-serializable. (You should also avoid providing a `:locations` value, as that is reserved per the GraphQL spec.)

Note there is currently issue listed to be able to provide multiple errors per field, but it's not high on our priority list at the moment. (PRs with tests, as always, are welcome.)

**Comment by ndarilek** [comment]
Ah, got it. So if anyone else finds this, I seem to have this working:

```
     field :get_open_order_count, type: :integer do
       resolve fn(args, %{context: context}) ->
         {:error, %{message: "bar"}}
       end
     end
```

Would it be possible for this error to specify that a `:message` key is 
required in the map value? To me it looks like the claim is that I 
didn't provide a map, which didn't make sense to me.

Thanks for the quick response.

**Comment by bruce** [comment]
Absolutely; in fact I'll just reopen this so I can make sure that, specifically, gets handled.

**Comment by benwilson512** [comment]
We should also update the typespecs to indicate that it has required keys.

**Comment by bruce** [comment]
This is getting cleaned up in #248.

**Comment by bruce** [comment]
Fix in `master`, will be in `v1.2.4`.
